index.ts

```diff
@@ -70,7 +69,7 @@ export function isHttps(requestUrl: string) {
 }
 
 export class HttpClient {
-    userAgent: string | null | undefined;
```

Comment:
good since I made constructor arg optional

Reply:
An optional parameter in typescript is `T | undefined` so the input won't be `null` if left out, it will be `undefined`.

See https://www.typescriptlang.org/docs/handbook/functions.html#optional-and-default-parameters

Reply:
OK